ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -216,6 +241,24 @@ private MessageWriteSetStateInStore checkWriteSetStateInStore(MessageWriteSet me
     }
   }
 
+  private void checkCapacityAndUpdateHelix(long totalCapacity, long usedCapacity) {
+    if (clusterManagerWriteStatusDelegate != null) {
+      double percentFilled = (index.getLogUsedCapacity() / (double) log.getCapacityInBytes()) * 100;
+      if (percentFilled > config.storeReadOnlyEnableSizeThresholdPercentage && writeState != WriteState.READ_ONLY) {
+        clusterManagerWriteStatusDelegate.seal(replicaId);
+        writeState = WriteState.READ_ONLY;
+      } else if (percentFilled
+          <= config.storeReadOnlyEnableSizeThresholdPercentage - config.storeReadWriteEnableSizeThresholdPercentageDelta
+          && writeState != WriteState.READ_WRITE
+          || percentFilled <= config.storeReadOnlyEnableSizeThresholdPercentage && writeState == WriteState.IDLE) {
+        clusterManagerWriteStatusDelegate.unseal(replicaId);
+        writeState = WriteState.READ_WRITE;
+      }
+    } else {
+      logger.warn("ClusterManagerWriteStatusDelegate not set, dynamic helix write status turned off");
```

Comment:
warn vs info?

Reply:
I had it at warn since it seems that should be a problem in production, but looking at it I think it should be placed in the initializer or constructor instead (I'm making the delegate reference final so one log line should be enough)